tests/unit/regression/krrnystrom_unittest.cc

```diff
@@ -0,0 +1,133 @@
+/*
+* Copyright (c) The Shogun Machine Learning Toolbox
+* Written (w) 2016 Fredrik Hallgren
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* 1. Redistributions of source code must retain the above copyright notice, this
+* list of conditions and the following disclaimer.
+* 2. Redistributions in binary form must reproduce the above copyright notice,
+* this list of conditions and the following disclaimer in the documentation
+* and/or other materials provided with the distribution.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+* ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*
+* The views and conclusions contained in the software and documentation are those
+* of the authors and should not be interpreted as representing official policies,
+* either expressed or implied, of the Shogun Development Team.
+*/
+
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/kernel/GaussianKernel.h>
+#include <shogun/regression/KRRNystrom.h>
+#include <shogun/regression/KernelRidgeRegression.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+
+/**
+ * Test that the expected interface is available.
+ */
+TEST(KRRNystrom, test_interface)
+{
+
+	/* Use default constructor */
+	CKRRNystrom krr;
+
+	/* Test setters */
+	krr.set_tau(0.1);
+	krr.set_m(100);
+}
+
+TEST(KRRNystrom, test_get_name)
+{
+	CKRRNystrom krr;
+	EXPECT_EQ(krr.get_name(), "KRRNystrom");
+}
+
+TEST(KRRNystrom, test_get_classifier_type)
+{
+	CKRRNystrom krr;
+	EXPECT_EQ(krr.get_classifier_type(), CT_KERNELRIDGEREGRESSION);
+}
+
+/**
+ * Test the main algorithm by comparison of the alphas and predictions
+ */
+TEST(KRRNystrom, compare_to_KRR)
```

Comment:
name the test so that it states that it compares using *all* columns
Then, there should be another test that uses only say half of the columns (with a lower accuracy)